### PR TITLE
Unify kafka client setup

### PR DESCRIPTION
**Pull request by jvlcek**
This PR will allow the kafka client and server be configured with an IP Address instead of, the Kafka favored, DNS resolvable hostname.

Fixes https://github.com/ManageIQ/manageiq-appliance_console/issues/135
Fixes https://github.com/ManageIQ/manageiq-appliance_console/issues/146

This issue is tracked here: [Investigate how to handle Kafka's apparent dependence on hostname #146](https://github.com/ManageIQ/manageiq-appliance_console/issues/146)

This PR will also combine the existing kafka messaging client for non-secure connections to a broker with the new secure kafka client configuration.

We will always configure out Kafka broker, with certificates, on secure port 9093. However we will support connecting, as a kafka client, to a kafka broker listening on unsecured port 9092.

This issue is tracked here: [Combine Kafka client configuration for external server and ManageIQ server #135](https://github.com/ManageIQ/manageiq-appliance_console/issues/135)

The matrix of supported cases is as follows

```
MiQ configured secure kafka Broker on port 9093 with DNS resolvable hostname  <-> MiQ configured kafka client
MiQ configured secure kafka Broker on port 9093 with IP Address               <-> MiQ configured kafka client
A remote, not MiQ, kafka Broker on port 9092 with DNS resolvable hostname     <-> MiQ configured kafka client
A remote, not MiQ, kafka Broker on port 9092 with IP Address                  <-> MiQ configured kafka client
```

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot add_label enhancement

**Comment by jvlcek** [comment]
@bdunne Please review

**Comment by jvlcek** [comment]
@miq-bot assign @bdunne

**Comment by jvlcek** [comment]
@Fryguy Please comment on the Hakiri failure. I don't see the connection to the changes introduced in this PR.

**Comment by Fryguy** [comment]
Ignore hakiri...I can't seem to turn it off for this repo

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (bdunne) [inline lib/manageiq/appliance_console/message_configuration.rb:90]: Is this supposed to be here?

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (bdunne) [inline lib/manageiq/appliance_console/message_configuration.rb:74]: Can this method leverage the unsecure method for the majority of the content?

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (bdunne) [inline lib/manageiq/appliance_console/message_configuration.rb:7]: `secure` and `server_host_is_ipaddr` shouldn't be listed here since they're methods, right?

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (bdunne) [inline lib/manageiq/appliance_console/message_configuration.rb:204]: Can this use `String#ipaddress?` from more_core_extensions instead?

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (bdunne) [inline lib/manageiq/appliance_console/message_configuration_server.rb:219]: Can this return just the `ks_params` and have the callers of `ks_alias` be `ks_params["-alias"]`?

Comment (bdunne) [reply lib/manageiq/appliance_console/message_configuration_server.rb:219]: Also, can `ks_params` be `keystore_params`?

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb:90]: Thank you for the feedback @bdunne  I appreciate it.
Yes `ssl.endpoint.identification.algorithm=#{algorithm}` is needed here to enable IPAddr resolution to work.

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb:74]: I'd be glad to discuss suggestions you may have but it took a lot of work to get the content of the client properties file content nailed down for the matrix of `hostname` vs `IP Addr` and `secure` vs `unsecure`.  I think leaving it as two separate fully complete methods may help future maintainers.

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb:7]: `secure?` and `server_host_is_ipadd?` are methods but
#deactive resets `secure` and `server_host_is_ipaddr` back to nil. So I think they need to be here.

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb:204]: Thank you. I will explore that.

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration_server.rb]: @bdunne 👍 
Addressed by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/0da11790549b1b5a6f97370e6eca2c5198b36751

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb]: Addressed by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/1438dd4126fec5988899d13c39f018c70e1c641c

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (Fryguy) [inline lib/manageiq/appliance_console/message_configuration_server.rb:194]: This feels like a potential command injection issue.  We shouldn't be passing params this way with AwesomeSpawn (yes, I see it was that way before), but it's an actual issue after this change in that there is potential user injection.

```suggestion
        AwesomeSpawn.run!("firewall-cmd", :params => {:remove_port => "#{server_port}/tcp", :permanent => nil}) # secure
```

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (Fryguy) [inline lib/manageiq/appliance_console/message_configuration_server.rb:114]: Same here with the command injection

```suggestion
        AwesomeSpawn.run!("firewall-cmd", :params => {:add_port => "#{server_port}/tcp", :permanent => nil)
```

**Comment by jvlcek** [comment]
Thank you @Fryguy 
The issues you raised are resolved with commit https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/9b05d5b046cb50a27dde8fa2dd01a37aba2621c9

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (bdunne) [reply lib/manageiq/appliance_console/message_configuration.rb:7]: `#deactivate` is setting `@secure = nil`, so you shouldn't need the `attr_reader`

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (Fryguy) [reply lib/manageiq/appliance_console/message_configuration.rb:74]: I think keeping them separate is fine for now.  I can see both side of the argument.

**Comment by Fryguy** [comment]
LGTM, but I admit not knowing much about this gem, so I defer to @bdunne

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb]: @bdunne This is resolved by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/14e2e666ddcd3b1c1cbebadc3334105853b954e3

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (Fryguy) [inline lib/manageiq/appliance_console/message_configuration.rb:196]: `||=` won't cache as expected here because you can't cache a boolean properly with `||=`  (because false will always try again).

If we do want to cache this, this should be
```suggestion
        return @secure if defined?(@secure)

        @secure = server_port == 9_093
```

Comment (Fryguy) [reply lib/manageiq/appliance_console/message_configuration.rb:196]: That being, said, I wouldn't cache this at all, unless it's expensive, and an `==` is not expensive. (Not sure if server_port is a regular accessor or something more complicated)

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (Fryguy) [inline lib/manageiq/appliance_console/message_configuration.rb:200]: Same here with caching a boolean, and also same question if this is even necessary.  If it's not necessary, then I'd argue even further on this one that the method might not be needed at all since the method name is practically identical to the code in question

```ruby
server_host_is_ipaddr?
#vs
server_host.ipaddress?
```

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (Fryguy) [inline lib/manageiq/appliance_console/message_configuration_server.rb:204]: Super minor, but this might be easier to read if broken up into multple lines.

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb]: Thank you @Fryguy!
Resolved by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/14c07ec105fc32ca6fdf36842456fcde4a54d0a0

File: lib/manageiq/appliance_console/message_configuration_server.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration_server.rb:204]: Resolved by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/149/commits/14c07ec105fc32ca6fdf36842456fcde4a54d0a0

File: lib/manageiq/appliance_console/message_configuration.rb
Comment (jvlcek) [reply lib/manageiq/appliance_console/message_configuration.rb]: These attr_readers have been removed.

**Review by bdunne — APPROVED** [review]

**Pull request merged**
